### DISCUSSION TITLE
Parallel builds from the same builder

### DISCUSSION THREAD

**Discussion by nlamprian** [buildbot/buildbot#6139] [Q&A]
I'm trying to figure out how to set up my Buildbot instance for running parallel builds from a single builder. Here is what I've gathered so far:

* Each worker runs one build at a time from a given builder
* If I have one worker per machine, I can run N(<=max_builds) parallel builds on that worker from N different builders
* If I want to limit the builds on a worker, I can use a worker lock
* If I want parallel builds from one builder, I need multiple workers
* If I have multiple workers, I cannot synchronize them per host; the worker lock is meant for a single worker, and there is no equivalent lock for a host

So here are my questions:
* Is it possible to run parallel builds from the same builder on a given worker?
* Is this configuration not supported by the architecture of Buildbot? I.e. do I need multiple builders instead?

**Comment by vibbo** [comment]
Hi, a bit late reaction, but I will try nevertheless.

AFAIK your assumptions are all correct with a caveat for the last one addressed in answer for the second question.

* Answer to the first question is yes - there is only one `workdir` for every builder on single worker.
* Second question depends on what you are trying to achieve. If you situation requires multiple parallel build on single host, you can use multiple workers on that hosts (as was one of your assumptions). If you want additional synchronization you could use `MasterLock`(s) in Steps (see [example](http://docs.buildbot.net/current/manual/configuration/interlocks.html?highlight=workerlock#examples)).
